crates/solidity/outputs/cargo/crate/src/main.rs

```diff
@@ -73,11 +73,11 @@ fn execute_parse_command(file_path_string: String, version: Version, json: bool)
         println!("{json}");
     }
 
-    if !errors.is_empty() {
-        std::process::exit(1);
+    if errors.is_empty() {
```

Comment:
This is different, as returning an error from main (for an internal error/crash/bug) will print a ton of stack traces, making it hard to actually see the syntax errors (output of the tool). If we got to this point, we just want to print the syntax errors and exit cleanly without panics/traces (hence the `exit(1)`).

Maybe we should add a comment here about this.

Reply:
You probably mean another path, where I already marked it as `allow` and it already has a comment about why we're doing it (omitting stack traces).

Here, we already print context on previous errors in `execute_parse_command` and I didn't change that; I only changed the `exit(1)` to `bail!`. Because this does not carry additional context, only the error message will be printed ("Couldn't parse the Solidity source file") and the resulting error code will be 1.

Reply:
But still, because it is an `anyhow::Error`, stack traces (for that single error) will be printed on the terminal. Right?

```bash
infra run slang_solidity parse --version 0.8.0 'crates/solidity/testing/snapshots/cst_output/Expression/_0/input.sol'
```

```log
Error: Expected ConstantKeyword or OpenBracket.
   ╭─[input.sol:1:3]
   │
 1 │ _0
   │   │ 
   │   ╰─ Error occurred here.
───╯
Error: Couldn't parse the Solidity source file.

Stack backtrace:
   0: backtrace::backtrace::libunwind::trace

..... a lot more traces......

```

Reply:
That is because we, by default, run the commands internally with `RUST_BACKTRACE=full`.

When executing it with a clean environment (e.g. `cargo run`), we get the plain error:
```
$ cargo run -p slang_solidity parse --version 0.8.0 'crates/solidity/testing/snapshots/cst_output/Expression/_0/input.sol'
    Finished dev [unoptimized + debuginfo] target(s) in 0.07s
     Running `target/debug/slang_solidity parse --version 0.8.0 crates/solidity/testing/snapshots/cst_output/Expression/_0/input.sol`
Error: Expected ConstantKeyword or OpenBracket.
   ╭─[crates/solidity/testing/snapshots/cst_output/Expression/_0/input.sol:1:3]
   │
 1 │ _0
   │   │ 
   │   ╰─ Error occurred here.
───╯
Error: Couldn't parse the Solidity source file.
```

Reply:
Yes, and a lot of Rust developers (or repos with Rust code will have that turned on).

But regardless. The CLI performed precisely as it is intended, delivering the required results (AST + errors), and now it is signaling the presence of errors via exit codes. This is not an internal failure. The stack traces we print are not relevant, useful, or actionable by the user.